[BUG] TS compile: "TS2456: Type alias 'Middleware' circularly references itself."

**Describe the bug**
Error thrown when compiling react app with typescript v4.9.4 and using yarn pnp.
[Example project](https://github.com/bobmazy/react-typescript)

`.yarn/__virtual__/react-tooltip-virtual-7b0a038807/0/cache/react-tooltip-npm-5.7.0-8f25b21000-1d3ae9afc9.zip/node_modules/react-tooltip/dist/react-tooltip.d.ts:16:6 - error TS2456: Type alias 'Middleware' circularly references itself.`


**Version of Package**
v5.6.0

**To Reproduce**
1. add typescript `v4.9.4` to react app 
2. use yarn pnp
3. use react-tooltip `v5.6.0`
4. compile

or 
1. clone https://github.com/bobmazy/react-typescript
2. run `yarn`
3. run `yarn tsc`

**Expected behavior**
React App should compile without any error

--- a/src/components/Tooltip/TooltipTypes.d.ts
+++ b/src/components/Tooltip/TooltipTypes.d.ts
@@ -1,5 +1,6 @@
 import type { ElementType, ReactNode, CSSProperties } from 'react'
-import type { Middleware } from '@floating-ui/dom'
+
+export type { Middleware } from '@floating-ui/dom'
 
 export type PlacesType = 'top' | 'right' | 'bottom' | 'left'
 
@@ -12,8 +13,6 @@ export type ChildrenType = Element | ElementType | ReactNode
 export type EventsType = 'hover' | 'click'
 
 export type PositionStrategy = 'absolute' | 'fixed'
-
-export type Middleware = Middleware
 
 export type DataAttribute =
   | 'place'

--- a/src/components/Tooltip/TooltipTypes.d.ts
+++ b/src/components/Tooltip/TooltipTypes.d.ts
@@ -1,6 +1,7 @@
 import type { ElementType, ReactNode, CSSProperties } from 'react'
+import type { Middleware } from '@floating-ui/dom'
 
-export type { Middleware } from '@floating-ui/dom'
+export type { Middleware }
 
 export type PlacesType = 'top' | 'right' | 'bottom' | 'left'
 

--- a/src/index.tsx
+++ b/src/index.tsx
@@ -8,6 +8,7 @@ import type {
   VariantType,
   WrapperType,
   IPosition,
+  Middleware,
 } from './components/Tooltip/TooltipTypes'
 import type { ITooltipController } from './components/TooltipController/TooltipControllerTypes'
 import type { ITooltipWrapper } from './components/TooltipProvider/TooltipProviderTypes'
@@ -26,4 +27,5 @@ export type {
   ITooltipController as ITooltip,
   ITooltipWrapper,
   IPosition,
+  Middleware,
 }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,6 +1,6 @@
 {
   "include": ["./global.d.ts", "./src/**/*.ts", "./src/**/*.js", "./src/**/*.tsx"],
-  "exclude": [],
+  "exclude": ["src/test/**/*"],
   "compilerOptions": {
     /* Visit https://aka.ms/tsconfig to read more about this file */
 
@@ -104,6 +104,6 @@
 
     /* Completeness */
     // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
-    "skipLibCheck": true /* Skip type checking all .d.ts files. */
+    "skipLibCheck": false /* Skip type checking all .d.ts files. */
   }
 }

